### PR TITLE
feat:updating go.mod dependencies for alpha4

### PR DESCRIPTION
**Pull request by chloeyin**
This PR updates notation cli with the main branch of `notation-core-go` and `notation-go`.
After all PRs merged, I will update go.mod to use main branch.
Test manually on virtual machine.
Now I hard code envelope type to be jws. I will add more envelope types after cose merged.
Signed-off-by: zaihaoyin <zaihaoyin@microsoft.com>

### CODE REVIEW

File: internal/cmd/signer.go [hunk 1]
```diff
@@ -13,9 +14,11 @@ import (
 // GetSigner returns a signer according to the CLI context.
 func GetSigner(opts *SignerFlagOpts) (notation.Signer, error) {
 	// Construct a signer from key and cert file if provided as CLI arguments
+	// TODO: support cose media type
+	enveLopeType := jws.MediaTypeEnvelope
 	if keyPath := opts.KeyFile; keyPath != "" {
 		certPath := opts.CertFile
-		return signature.NewSignerFromFiles(keyPath, certPath)
+		return signature.NewSignerFromFiles(keyPath, certPath, enveLopeType)
```
Comment (patrickzheng200) [inline internal/cmd/signer.go:21]: nit: envelopeType

File: internal/cmd/signer.go [hunk 2]
```diff
@@ -24,7 +27,7 @@ func GetSigner(opts *SignerFlagOpts) (notation.Signer, error) {
 		return nil, err
 	}
 	if key.X509KeyPair != nil {
-		return signature.NewSignerFromFiles(key.X509KeyPair.KeyPath, key.X509KeyPair.CertificatePath)
+		return signature.NewSignerFromFiles(key.X509KeyPair.KeyPath, key.X509KeyPair.CertificatePath, enveLopeType)
```
Comment (patrickzheng200) [inline internal/cmd/signer.go:30]: same here

File: internal/cmd/signer.go [hunk 3]
```diff
@@ -34,7 +37,7 @@ func GetSigner(opts *SignerFlagOpts) (notation.Signer, error) {
 		if err != nil {
 			return nil, err
 		}
-		return signature.NewSignerPlugin(runner, key.ExternalKey.ID, key.PluginConfig)
+		return signature.NewSignerPlugin(runner, key.ExternalKey.ID, key.PluginConfig, enveLopeType)
```
Comment (patrickzheng200) [inline internal/cmd/signer.go:40]: same here

**Review by binbin-li — APPROVED** [review]
lgtm

**Review by patrickzheng200 — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**